learning_resources_search/views.py

```diff
@@ -116,10 +135,10 @@ def get_queryset(self):
 
     @extend_schema(
         summary="Subscribe user to query",
-        request=LearningResourcesSearchRequestSerializer(),
+        request={"multipart/form-data": LearningResourcesSearchRequestSerializer()},
```

Comment:
curious...does it need to use form-data? I feel like most of our apis use json.

Reply:
I would have liked to have this as json but I was encountering issues in getting the specs and the autogenerated apis and how the params were sent over without it.

Reply:
Would be best to avoid this if possible.  

I tried with this:
```python
    @extend_schema(
        summary="Subscribe user to query",
        parameters=[LearningResourcesSearchRequestSerializer],
        request=LearningResourcesSearchRequestSerializer(),
        responses=PercolateQuerySerializer(),
    )
    @action(detail=False, methods=["post"], name="Subscribe user to query")
```    

and was able to post `{"q": "physics"}`  via json at this [url ](http://localhost:8063/api/v1/learning_resources_user_subscription/subscribe/ )successfully.

I think the swagger schema spec is also correct (screenshot below) - assuming it should be identical to a search request?  Not sure about the autogenerated frontend specs though, is that where the main problem is?

![Screenshot 2024-05-13 134227](https://github.com/mitodl/mit-open/assets/187676/a7c29aea-e753-49fe-8776-0bb989c65a3c)

Reply:
The verbosity of this has always annoyed me, but the JSON version would work fine with 
```ts
      onClick={() =>
        subscriptionCreate.mutateAsync({
          LearningResourcesSearchRequestRequest: subscribeParams,
        })
      }
```

Reply:
I'm also not sure why it's `SearchRequestRequest` (2x Request). That seems like an issue with our naming or something. I would have expected "just" `LearningResourcesSearchRequest`

Reply:
thanks! those two changes did appear to fix the json api.